filesystem/android_super.ksy

```diff
@@ -0,0 +1,196 @@
+meta:
+  id: android_super
+  title: Android Dynamic Partitions metadata
+  application: Android
+  file-extension: img
+  license: CC0-1.0
+  ks-version: 0.9
+  bit-endian: le
+  endian: le
+
+doc: |
+  The metadata stored by Android at the beginning of a "super" partition, which
+  is what it calls a disk partition that holds one or more Dynamic Partitions.
+  Dynamic Partitions do more or less the same thing that LVM does on Linux,
+  allowing Android to map ranges of non-contiguous extents to a single logical
+  device. This metadata holds that mapping.
+
+doc-ref:
+  - https://source.android.com/devices/tech/ota/dynamic_partitions/implement
+  - https://android.googlesource.com/platform/system/core/+/master/fs_mgr/liblp/include/liblp/metadata_format.h
+
+seq:
+  - id: padding
+    size: 0x1000
+    type: dummy
+
+  - id: primary_geometry
+    size: 0x1000
+    type: geometry
+
+  - id: backup_geometry
+    type: geometry
+    size: 0x1000
+
+  - id: primary_metadata
+    size: primary_geometry.metadata_max_size
+    type: metadata
+    repeat: expr
+    repeat-expr: primary_geometry.metadata_slot_count
+
+  - id: backup_metadata
+    size: primary_geometry.metadata_max_size
+    type: metadata
+    repeat: expr
+    repeat-expr: primary_geometry.metadata_slot_count
+
+types:
+  dummy: {}
+
+  geometry:
+    seq:
+      - id: magic
+        contents: 'gDla'
+      - id: struct_size
+        type: u4
+      - id: checksum
+        size: 32
```

Comment:
Is the checksum format known? Is it an output from [known hash function](https://en.wikipedia.org/wiki/List_of_hash_functions), or something custom?

Reply:
It's SHA-256. The header checksum is computed with a checksum field of all zeroes. Want me to document that inline? The second top-level `doc-ref` URL mentions it.

Reply:
> It's SHA-256. The header checksum is computed as if the checksum field were all zeroes. Want me to document that inline? The second top-level `doc-ref` URL mentions it.

Definitely a good idea, put this description to the `doc` key of this attribute, please.

Reply:
Updated.